akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1989,6 +1989,15 @@ trait FlowOpsMat[+Out, +Mat] extends FlowOps[Out, Mat] {
   def mapMaterializedValue[Mat2](f: Mat ⇒ Mat2): ReprMat[Out, Mat2]
 
   /**
+    * Materializes to `FlowMonitor[Out]` that allows monitoring of the the current flow. All events are propagated
+    * by the monitor unchanged. Note that the monitor inserts a memory barrier every time it processes an
+    * event, and may therefor affect performance.
+    * The `combine` function is used to combine the `FlowMonitor` with this flow's materialized value.
+    */
+  def monitor[Mat2]()(combine: (Mat, FlowMonitor[Out]) => Mat2): ReprMat[Out, Mat2] =
```

Comment:
We should update the comment though as it is not very helpful (it can be interpreted in various ways):
```
INTERNAL API: extending this trait is not supported under the binary compatibility rules for Akka.
```